### PR TITLE
Add table aws_directory_service_certificate Closes #1835

### LINKED ISSUE #1835: Add table aws_directory_service_certificate
**References**
Add table `aws_directory_service_certificate`
List API reference: https://docs.aws.amazon.com/directoryservice/latest/devguide/API_ListCertificates.html
Get API reference: https://docs.aws.amazon.com/directoryservice/latest/devguide/API_DescribeCertificate.html

### PR DESCRIPTION
**Pull request by ParthaI**
# Integration test logs
<details>
  <summary>Logs</summary>

```
Add passing integration test logs here
```
</details>

# Example query results
<details>
  <summary>Results</summary>

```
> select * from aws_aab.aws_directory_service_certificate
+--------------+----------------+-------------+-------------+------------+---------------------------+---------------------------+--------------------------------------+---------------------------+------>
| directory_id | certificate_id | common_name | type        | state      | expiry_date_time          | registered_date_time      | state_reason                         | client_cert_auth_settings | title>
+--------------+----------------+-------------+-------------+------------+---------------------------+---------------------------+--------------------------------------+---------------------------+------>
| d-90679c9511 | c-9067387b7a   | unused      | ClientLDAPS | Registered | 2024-05-16T14:07:09+05:30 | 2023-07-18T13:22:51+05:30 | Certificate registered successfully. | <null>                    | c-906>
+--------------+----------------+-------------+-------------+------------+---------------------------+---------------------------+--------------------------------------+---------------------------+-----
```
</details>

### CODE REVIEW

File: aws/table_aws_directory_service_certificate.go [hunk 1]
```diff
@@ -0,0 +1,253 @@
...
+			},
+			Hydrate: getDirectoryServiceCertificate,
+		},
+		List: &plugin.ListConfig{
+			ParentHydrate: listDirectoryServiceDirectories,
+			Hydrate:       listDirectoryServiceCertificates,
+			KeyColumns: []*plugin.KeyColumn{
+				{
+					Name:    "directory_id",
+					Require: plugin.Optional,
```
[the remark this answers is not in the document]
Comment (ParthaI) [reply aws/table_aws_directory_service_certificate.go:42]: @madhushreeray30, No, this should be optional quals are awe are using the directory as our parent hydrate.

We can minimize the number of API call by providing it as optional quals.

[the remark this answers is not in the document]
Comment (madhushreeray30) [reply aws/table_aws_directory_service_certificate.go:42]: ok @ParthaI

File: aws/table_aws_directory_service_certificate.go [hunk 2]
```diff
@@ -0,0 +1,256 @@
...
+		Limit:       aws.Int32(maxLimit),
+	}
+
+	pagesLeft := true
+	// List call
+	for pagesLeft {
+		result, err := svc.ListCertificates(ctx, input)
+		if err != nil {
+			// Should we ignore this error?
+			// In the case of parent hydrate the ignore config is not sesms to be working fine. So we need handle it manually
```
Comment (madhushreeray30) [inline aws/table_aws_directory_service_certificate.go:163]:
```suggestion
			// In the case of parent hydrate the ignore config seems to not work fine. So we need to handle it manually```

File: docs/tables/aws_directory_service_certificate.md [hunk 3]
```diff
@@ -0,0 +1,81 @@
...
+  certificate_id,
+  common_name,
+  type,
+  state,
+  expiry_date_time
+from
+  aws_directory_service_certificate;
+```
+
+### List MicrosoftAD type directories
```
Comment (madhushreeray30) [inline docs/tables/aws_directory_service_certificate.md:25]:
```suggestion
### List 'MicrosoftAD' type directories
```

File: docs/tables/aws_directory_service_certificate.md [hunk 4]
```diff
@@ -0,0 +1,81 @@
...
+  d.name as directory_name,
+  d.type as directory_type
+from
+  aws_directory_service_certificate c,
+  aws_directory_service_directory d
+where
+  d.type = 'MicrosoftAD';
+```
+
+### List certificates that are deregistered
```
Comment (madhushreeray30) [inline docs/tables/aws_directory_service_certificate.md:42]:
```suggestion
### List deregistered certificates
```

File: docs/tables/aws_directory_service_certificate.md [hunk 5]
```diff
@@ -0,0 +1,81 @@
...
+  directory_id,
+  certificate_id,
+  common_name,
+  type,
+  state,
+  expiry_date_time
+from
+  aws_directory_service_certificate
+where
+  expiry_date_time >= time() - interval '30' day;
```
Comment (madhushreeray30) [inline docs/tables/aws_directory_service_certificate.md:69]: It is more useful to list certificated that will expire soon 
```suggestion
### List certificates that will expire in the coming 7 days

```sql
select
  directory_id,
  certificate_id,
  common_name,
  type,
  state,
  expiry_date_time
from
  aws_directory_service_certificate
where
  expiry_date_time >= now() + interval '7' day;
```

File: docs/tables/aws_directory_service_certificate.md [hunk 6]
```diff
@@ -0,0 +1,81 @@
...
+```
+
+### Get client certificate auth settings of each certificate
+
+```sql
+select
+  directory_id,
+  certificate_id,
```
Comment (madhushreeray30) [inline docs/tables/aws_directory_service_certificate.md:77]:
```suggestion
  certificate_id,
  common_name,
```

**Review by madhushreeray30 — COMMENTED** [review]
@ParthaI please have a look at the comments, thanks!

**Review by madhushreeray30 — APPROVED** [review]
LGTM 👍

**Review by misraved — APPROVED** [review]

**Pull request merged**